panel/io/cache.py

```diff
@@ -414,7 +452,7 @@ def hash_func(*args, **kwargs):
 
     if iscoroutinefunction(func):
         @functools.wraps(func)
-        async def wrapped_func(*args, **kwargs):
+        async def wrapped_func(*args: P.args, **kwargs: P.kwargs) -> R:
```

Comment:
For me to learn. Can you explain what goes on here? Thanks.

Reply:
`ParamSpec` `P` represents the type signatures of the `func` that was passed in. Since `func` was annotated as `Callable[P, R]`, this is just saying that `wrapped_func` has the same signature as `func`. It doesn't really make a practical difference here, but for instance, something you could do is add an extra positional or keyword argument to `wrapped_func`, and that would get validated in subsequent calls.